### PR TITLE
CSV Exercise

### PR DESCRIPTION
**Pull request by CheezItMan**
A CSV Practice Activity and later students will use the CSV data with enumerables to manipulate the given data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# CSV & Enumerables Practice
+
+In this exercise you are asked to process a CSV file with Olymic Games data.  This exercise has two parts.  The first allowing you to practice reading in data from a CSV file, processing it and writing the results back to a file.  The second involves using Enumerable methods to manipulate the data from the file.
```
Comment (tildeee) [inline README.md]: "Olymic Games" -> "Olympic Games" typo

Comment (tildeee) [reply README.md]: Grammar: "The first allowing you to ..." -> "The first part allows you to" or something similar, like, "In the first part, you will practice reading in data ..."

**Review by tildeee — CHANGES_REQUESTED** [review]
I LOVE THIS! Particularly the tests you provided.

Also it's a great exercise; just implement three methods. It's great!

If this is a one-hour assignment, I actually would consider making some of Part 2 optional?

Lastly, you have a stray lock file in here: `data/.~lock.Sacramento...` to delete. Otherwise, I think it's fine to merge

**Comment by CheezItMan** [comment]
The idea is that it's a 2 parter with more practice using Enums later in the week.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+-   `total_medals_per_country(olympic_data)`  - This method takes the array of hashes read in from the file in `load_data` and returns an array of hashes with each element having a country name (team), and number of medals that country (team) has won.
+-   `save_medal_totals(filename, medal_totals)` - This method writes the medal total information returned from `total_medals_per_country` to another CSV file provided by `filename`.
+
+## Part 2:
+
+In part 2 you will create methods to manipulate the data read in from the file.  You should think carefully and use an Enumerable method to help you complete the method.
+
+-   `all_gold_medal_winners(olympic_data)` - This method returns an array the names of all Gold metal winners.
+-   `metals_sorted_by_country(metal_totals)`  - This method takes a list of countries and their metal totals and returns the same list, sorted by country
+-    `country_with_most_metals(metal_totals)` - This method returns the country with the highest number of metals.
```
Comment (droberts-sea) [inline README.md]: These should all be `medal` not `metal`

**Review by droberts-sea — CHANGES_REQUESTED** [review]
Looks good, aside from spelling.

**Review by Hamled — COMMENTED** [review]
I think this looks good, with the changes suggested already by @tildeee and @droberts-ada.

One suggestion I would make for the README is to call out more specifically the data structures that will be used as input for each method.

You do a good job of making it clear that in Part 1 the input for `total_medals_per_country` is the output from `load_data` (and similarly for `save_medal_totals`). However, I think it would also be good to point them to the test suite because it also contains a description of the structure that `load_data` should return.

And for Part 2 it seems like, based on the parameter names, those input data structures are the same as the ones used/created in Part 1. If that is true then I think it would help to call that out more explicitly (e.g. saying that `medals_sorted_by_country` takes the same input as what `total_medals_per_country` returns).

**Pull request merged**